pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -146,8 +146,10 @@ func (r *DNSPolicyReconciler) buildClusterGateway(ctx context.Context, downstrea
 	singleClusterAddresses := make([]gatewayv1beta1.GatewayAddress, len(clusterAddress))
 
 	mc := &clusterv1.ManagedCluster{}
-	if err := r.Client().Get(ctx, client.ObjectKey{Name: downstreamClusterName}, mc, &client.GetOptions{}); err != nil {
-		return target, err
+	if downstreamClusterName != dns.NoneCluster {
```

Comment:
nit: NoneCluster reads awkwardly, what are we trying to convey? Is it that there is no downstream cluster or that we are in a single cluster?

Reply:
It will be replaced with the gateway instead